python/ray/data/_internal/output_buffer.py

```diff
@@ -1,17 +1,32 @@
-from typing import Any
+from typing import Any, Optional
+from dataclasses import dataclass
 
 from ray.data._internal.delegating_block_builder import DelegatingBlockBuilder
 from ray.data.block import Block, BlockAccessor, DataBatch
-from ray.data.context import MAX_SAFE_BLOCK_SIZE_FACTOR
+from ray.data.context import MAX_SAFE_BLOCK_SIZE_FACTOR, MAX_SAFE_ROWS_PER_BLOCK_FACTOR
+
+
+@dataclass
+class OutputBlockSizeOption:
+    target_max_block_size: Optional[int] = None
+    target_max_rows_per_block: Optional[int] = None
+
+    def __post_init__(self) -> None:
+        assert not (
+            self.target_max_block_size is not None
+            and self.target_max_rows_per_block is not None
+        ), "Only one of target_max_block_size or target_max_rows_per_block should be set."
```

Comment:
they cannot be both None either. 
This is a cleaner way to check one is None and the other is non-None. `assert (x is None) != (y is None)`